### PR TITLE
Scrapper, Daria Kapitanova - 21FPL1

### PR DESCRIPTION
**Pull request by kapitanovadaria**

### CODE REVIEW

File: lab_5_scrapper/scrapper.py [hunk 1]
```diff
@@ -13,70 +71,120 @@ def __init__(self, path_to_config: Path) -> None:
...
         are not corrupt
         """
-        pass
+        seed_urls = self.config_content.seed_urls
+        total_articles = self.config_content.total_articles
+        headers = self.config_content.headers
+        encoding = self.config_content.encoding
+        timeout = self.config_content.timeout
+        should_verify_certificate = self.config_content.should_verify_certificate
+        headless_mode = self.config_content.headless_mode
```
Comment (dmitry-uraev) [inline lab_5_scrapper/scrapper.py]: can you avoid creating all these variables and validate config directly?

File: lab_5_scrapper/scrapper.py [hunk 2]
```diff
@@ -90,25 +198,52 @@ def __init__(self, config: Config) -> None:
...
+        self.urls = []
 
     def _extract_url(self, article_bs: BeautifulSoup) -> str:
         """
         Finds and retrieves URL from HTML
         """
-        pass
+        url = article_bs.get('href')
+        if isinstance(url, str) and url.startswith('/news/'):
+            new_url = 'https://informpskov.ru' + url
```
Comment (dmitry-uraev) [inline lab_5_scrapper/scrapper.py]: you can return result directly

File: lab_5_scrapper/scrapper.py [hunk 3]
```diff
@@ -90,25 +198,52 @@ def __init__(self, config: Config) -> None:
...
-        pass
+        seed_urls = self.get_search_urls()
+        for seed_url in seed_urls:
+            response = make_request(seed_url, self._config)
+            if response.status_code != 200:
+                continue
+            article_bs = BeautifulSoup(response.text, 'lxml')
+            article_page_bs = article_bs.find('div', {'id': 'yw0'})
+            article_page = article_page_bs.find_all(
+                'a')
```
Comment (dmitry-uraev) [inline lab_5_scrapper/scrapper.py]: why not write it on one line?

File: lab_5_scrapper/scrapper.py [hunk 4]
```diff
@@ -144,21 +285,35 @@ def parse(self) -> Union[Article, bool, list]:
...
     """
-    pass
+    prepare_environment(ASSETS_PATH)
+    config = Config(path_to_config=CRAWLER_CONFIG_PATH)
+    crawler = Crawler(config=config)
+    crawler.find_articles()
+    for i, full_url in enumerate(crawler.urls, start=1):
+        parser = HTMLParser(full_url=full_url, article_id=i, config=config)
+        article = parser.parse()
+        if isinstance(article, Article):
```
Comment (dmitry-uraev) [inline lab_5_scrapper/scrapper.py:343]: why do you use this condition here?

**Review by dmitry-uraev — CHANGES_REQUESTED** [review]
Some comments to respond to and we are done

File: lab_5_scrapper/scrapper.py (hunk 4 above)
Comment (kapitanovadaria) [reply lab_5_scrapper/scrapper.py:343]: code style won't pass if we don't check if article is an Article instance

File: lab_5_scrapper/scrapper.py (hunk 1 above)
Comment (kapitanovadaria) [reply lab_5_scrapper/scrapper.py]: done

File: lab_5_scrapper/scrapper.py (hunk 2 above)
Comment (kapitanovadaria) [reply lab_5_scrapper/scrapper.py]: done

File: lab_5_scrapper/scrapper.py (hunk 3 above)
Comment (kapitanovadaria) [reply lab_5_scrapper/scrapper.py]: done

**Review by dmitry-uraev — APPROVED** [review]
I think we are done here
Please proceed with lab 6

**Pull request closed without merging**